### PR TITLE
Structure output directory

### PR DESCRIPTION
**Pull request by pnmadelaine**
Remove the `--output-dir` option of `cargo hax into` and instead extract to `{crate}/hax/{backend}/extraction`

### CODE REVIEW

File: cli/driver/Cargo.toml [hunk 1]
```diff
@@ -25,3 +25,4 @@ log.workspace = true
 pretty_env_logger.workspace = true
 inquire = "0.6"
 const_format = "0.2"
+cargo_metadata = "0.15"
```
Comment (W95Psp) [inline cli/driver/Cargo.toml:28]: Can you move that dependency to the root `Cargo.toml`, and then use `cargo_metadata.workspace = true` here and in `test-harness/Cargo.toml`?

File: cli/driver/src/exporter.rs [hunk 2]
```diff
@@ -320,29 +320,34 @@ impl Callbacks for ExtractionCallbacks {
...
-                                    ))
-                                })
-                            }
-                        }
+                    let metadata = cargo_metadata::MetadataCommand::new().exec().unwrap();
+                    let crate_name = std::env::var("CARGO_CRATE_NAME").unwrap();
+                    let package = metadata
+                        .packages
+                        .iter()
+                        .rfind(|pkg| pkg.name == crate_name)
```
Comment (W95Psp) [inline cli/driver/src/exporter.rs:328]: why not just find?

**Review by W95Psp — APPROVED** [review]
Thanks :) jsut a few things

**Comment by W95Psp** [comment]
bors cancel

**Review by franziskuskiefer — APPROVED** [review]
The output structure needs to be described somewhere (Readme or book or something). Otherwise this looks god to me.

**Pull request merged**